package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java

```diff
@@ -1380,68 +1385,124 @@ private static List<Package> getFilesystemPackages(final PackageType type) {
+                wfOutputJson.add(key, gson.fromJson(value, JsonObject.class));
+            }
+            // write the workflow output
+            String outputString = gson.toJson(wfOutputJson);
+            if (Installer.endsWithIgnoreCase(wfOutputFilePath, "yaml")) {
+                outputString = Installer.convertJsonObjectToYaml(wfOutputFilePath, wfOutputJson);
+            }
+
+            Files.write(Paths.get(wfOutputFilePath), outputString.getBytes(), StandardOpenOption.CREATE);
+            // write the workflow error in a workflow error file (if any)
+            if (!StringUtils.isEmpty(workflowExecutionResult.getError())) {
+                Files.write(Paths.get(wfErrFilePath), workflowExecutionResult.getError().getBytes(), StandardOpenOption.CREATE);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (JsonSyntaxException e) {
+            throw new RuntimeException("Unable to parse input / output file data: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (WorkflowExecutionException e) {
+            try {
+                Files.write(Paths.get(wfErrFilePath), e.getMessage().getBytes(), StandardOpenOption.CREATE);
+            } catch (IOException e1) {
+                throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+            }
+            throw new RuntimeException("Workflow execution failed: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (Exception e) {
+            throw new RuntimeException("General error during workflow execution: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        }
+    }
+	
+    private static boolean isValidYaml(String yamlString) {
+        Yaml yaml = new Yaml();
+        try {
+            yaml.load(yamlString);
+            return true;
+        } catch (ParserException e) {
+            return false;
+        }
+    }
+
+    private static String convertYamlToJson(String fileName, String yamlString) {
+        Yaml yaml = new Yaml();
+        ObjectMapper jsonMapper = new ObjectMapper();
+        jsonMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        try {
+            // Parse YAML
+            Map<String, Object> yamlMap = yaml.load(yamlString);
+            // Convert to JSON
+            return jsonMapper.writeValueAsString(yamlMap);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(String.format("Unable to convert YAML file '%s' to JSON: '%s'", fileName, e.getLocalizedMessage()));
+        }
+    }
+
+    private static String convertJsonObjectToYaml(String fileName, JsonObject jsonObject) {
+        if (jsonObject.isEmpty()) {
+            return "--";
```

Comment:
Just a question, why do we return this string here? shouldn't we return empty string or throw an error?

Reply:
`---` is a separator in yaml,  but it's 3 `-`'s

Reply:
Thanks for the clarification, code updated.